### PR TITLE
Break up agent & server requirements

### PR DESCRIPTION
**Pull request by portante**
We now keep separate agent and server requirements files so that we can reduce the module dependencies when separately installed.

### CODE REVIEW

File: agent/requirements.txt [hunk 1]
```diff
@@ -1,15 +1,9 @@
...
-boto3
 bottle
 cffi
 click
 colorlog
-elasticsearch1==1.10.0
-flask
-flask-restful
-gunicorn
 humanize
```
Comment (dbutenhof) [inline agent/requirements.txt:6]: Are we using `humanize` in the agent? I thought Nikhil just added it for the server?

**Review by dbutenhof — APPROVED** [review]
Nice.

**Review by riya-17 — APPROVED** [review]

**Comment by zulcss** [comment]
lgtm

**Review by ndokos — APPROVED** [review]
LGTM

**Review by npalaska — DISMISSED** [review]
Looks good

File: agent/requirements.txt (hunk 1 above)
Comment (portante) [reply agent/requirements.txt:6]: I'll remove this.

File: tox.ini [hunk 2]
```diff
@@ -57,26 +62,41 @@ commands =
 
 [testenv:util-scripts]
 description = Runs all non-Python3-based agent util-scripts unit/functional tests
+deps =
+   -r{toxinidir}/agent/requirements.txt
+   -r{toxinidir}/test-requirements.txt
```
[the remark this answers is not in the document]
Comment (portante) [reply tox.ini:67]: I'll take a stab at doing that.

File: agent/requirements.txt (hunk 1 above)
Comment (portante) [reply agent/requirements.txt]: Removed.

File: tox.ini (hunk 2 above)
[the remark this answers is not in the document]
Comment (portante) [reply tox.ini]: This effort was worth it, because it removed a number of modules we were using for testing.

**Comment by portante** [comment]
> Btw, do we not have to update the rpm spec file to grab the correct requirements file when installing?

Once we merge this on `master`, we'll then update the `rpm-work` branch to take advantage of the separate `requirements.txt` files.

**Review by ndokos — APPROVED** [review]

**Review by npalaska — APPROVED** [review]
looks good, definitely helpful going forward

**Pull request merged**